watertap/costing/costing_base.py

```diff
@@ -35,6 +35,188 @@ class WaterTAPCostingBlockData(FlowsheetCostingBlockData):
+                        calculating annual water production
+            name (optional) - name for the annual water productionvariable
+                              Expression (default: annual_water_production)
+        """
+        self.add_component(
+            name,
+            pyo.Expression(
+                expr=(
+                    pyo.units.convert(
+                        flow_rate, to_units=pyo.units.m**3 / self.base_period
+                    )
+                    * self.utilization_factor
+                ),
+                doc=f"Annual water production based on flow {flow_rate.name}",
+            ),
+        )
+
+    def add_electricity_intensity(self, flow_rate, name="electricity_intensity"):
+        """
+        Add calculation of overall electricity intensity to costing block.
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating electricity intensity
+            name (optional) - the name of the Expression for the specific
+                              electrical intensity (default: specific_electrical_carbon_intensity)
+        """
+        self.add_specific_energy_consumption(flow_rate, name=name)
+
+    def add_specific_electrical_carbon_intensity(
+        self, flow_rate, name="specific_electrical_carbon_intensity"
+    ):
+        """
+        Add specific electrical carbon intensity (kg_CO2eq/m**3) to costing block.
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating specific electrical carbon intensity
+            name (optional) - the name of the Expression for the specific
+                              carbon intensity (default: specific_electrical_carbon_intensity)
+        """
+
+        self.add_component(
+            name,
+            pyo.Expression(
+                expr=self.aggregate_flow_electricity
+                * self.electrical_carbon_intensity
+                / pyo.units.convert(
+                    flow_rate, to_units=pyo.units.m**3 / pyo.units.hr
+                ),
+                doc=f"Specific electrical carbon intensity based on flow {flow_rate.name}",
+            ),
+        )
+
+    def _build_common_process_costs(self):
+        """
+        Build the common process costs to WaterTAP Costing Packages.
+        The currency units should already be registered.
+
+        The derived class should add constraints for total_capital_cost
+        and total_operating_cost
+        """
+        self.total_capital_cost = pyo.Var(
+            initialize=0,
+            doc="Total capital cost of the process",
+            units=self.base_currency,
+        )
+        self.total_operating_cost = pyo.Var(
+            initialize=0,
+            doc="Total operating cost of process per operating period",
+            units=self.base_currency / self.base_period,
+        )
+
+    def _build_common_global_params(self):
```

Comment:
So replaced `practical_investment_factor` with  `TIC`? I'd be good with that.

Reply:
They both still exist -- would need further refinement to eliminate one or the other.